lambda_backend/predict_microservice/service_inference.py

```diff
@@ -0,0 +1,82 @@
+from base_inference import BaseInference
+
+class ServiceInference(BaseInference):
+    """
+    Inference class for predictions on predict_microservice
+    
+    ...
+
+    Methods
+    -------
+    run(img, width_dict)
+        Obtains predicted boxes for predict_microservice
+    """
+
+    def run(self, img, width_dict):
```

Comment:
Validation of `numpy` matrix as input too

Reply:
I believe the input for both `handler.py` and `local_inference.py` is `cv2.Mat`, from [`cv2.imdecode`](https://docs.opencv.org/3.4/d4/da8/group__imgcodecs.html#ga26a67788faa58ade337f8d28ba0eb19e) and [`cv2.imread`](https://docs.opencv.org/4.5.2/d4/da8/group__imgcodecs.html#ga288b8b3da0892bd651fce07b3bbd3a56) respectively. [`cv2.dnn.blobFromImage` does accept a generic `InputArray` as `image`.](https://docs.opencv.org/3.4/d6/d0f/group__dnn.html#ga29f34df9376379a603acd8df581ac8d7) 

What should the validation here be for exactly?

Reply:
`InputArray` should be an interface for `Mat`. Since python `numpy.ndarray` is a binding to openCv `Mat` type. The validation should be `numpy.ndarray`. I think you can check with `print(type(img))`